02_commands.html

```diff
@@ -1,206 +1,524 @@
-                is not empty <var>value</var> will contain the error message describing why there was a failure.</li>
-          </ul>
-      </ol>
-
-      <h4 id='returning-http-response'>Returning a HTTP response to the local end</h4>
-      <p>when the command has completed encode the result using the following algorithm:</p>
-
-      <ol>
-        <li>Set the HTTP response status code to 200 unless described differently in the commands below</li>
-        <li>Let <var>sessionId</var> be the unique identifier for the current session</li>
-        <li>Add the contents of <var>result</var> to an object could be <a href='http://www.ecma-international.org/ecma-262/5.1/#sec-15.12.3'>stringified in to JSON</a>
-        <li>Let the payload body of the response be the <a href='http://www.ecma-international.org/ecma-262/5.1/#sec-15.12.3'>stringified JSON</a>
-          object.
-          <ol>
-            <li>If <a href="#newsession">newSession()</a> is called then the response payload body would look like the following
-              <pre>
-                {
-                  "sessionId" : "string value",
-                  "value" : capabilities-object,
-                }.
-              </pre>
-            </li>
-            <li>
-              If the WebDriver Command has to return data it should be the whole of the response payload body and look like the following
-              <pre>
-                {
-                  "value": "data from WebDriver Commands"
-                }
-              </pre>
-            </li>
-            <li>
-              If the WebDriver Command results in an error then the response payload body would look like the following
-              <pre>
-                {
-                  "status": "string value",
-                  "value" : "string message"
-                }
-              </pre>
-            </li>
-          </ol>
-        </li>
-        <li>Set HTTP headers for the follow keys and their values.
-          <ul>
-            <li><code>cache-control: no-cache</code></li>
-            <li><code>content-type: application/json</code></li>
-          </ul>
-        </li>
-        <li>Return HTTP response to the Local End</li>
-      </ol>
-    </section>
+    <h3>Introduction</h3>
+    <p>TODO</p>
   </section>
   <section>
-    <h3>Status Codes</h3>
-    <p>The WebDriver API indicates the success or failure of a command invocation via <var>status</var>. The following values are used and have the following meanings.</p>
+    <h3>Common Infrastructure</h3>
+    <h4>Algorithms</h4>
+    <p>Various parts of this specification are written in terms of step-by-step algorithms. The details of these algorithms do not have any normative significance; implementations are free to adopt any implementation strategy that produces equivalent output to the specification. In particular algorithms in this document are optimised for readability rather than performance.</p>
```

Comment:
No change necessary here, but:  It seems this applies to algorithms also in other chapters apart from the this about protocol.  We should consider uplifting this to a separate chapter on _Infrastructure_ in the future.